web/src/components/MoreOptionsDropdown.tsx

```diff
@@ -0,0 +1,215 @@
+import * as DropdownMenu from '@radix-ui/react-dropdown-menu';
+import { useFeatureFlag } from 'features/feature-flags/api';
+import { useShare } from 'hooks/useShare';
+import { useAtomValue } from 'jotai';
+import { Link, ShareIcon } from 'lucide-react';
+import { useMemo, useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import { FaFacebook, FaLinkedin, FaReddit, FaSquareXTwitter } from 'react-icons/fa6';
+import { twMerge } from 'tailwind-merge';
+import { ShareType, trackShareChart } from 'utils/analytics';
+import { baseUrl, DEFAULT_ICON_SIZE, DEFAULT_TOAST_DURATION } from 'utils/constants';
+import { hasMobileUserAgent as hasMobileUA } from 'utils/helpers';
+import { displayByEmissionsAtom, isHourlyAtom } from 'utils/state/atoms';
+
+import { DefaultCloseButton } from './DefaultCloseButton';
+import { TimeDisplay } from './TimeDisplay';
+import { Toast, useToastReference } from './Toast';
+
+// TODO: add chartId to tracking
+const onTrackShareChartReddit = trackShareChart(ShareType.REDDIT);
+const onTrackShareChartTwitter = trackShareChart(ShareType.TWITTER);
+const onTrackShareChartLinkedin = trackShareChart(ShareType.LINKEDIN);
+const onTrackShareChartFacebook = trackShareChart(ShareType.FACEBOOK);
+const onTrackShareChart = trackShareChart(ShareType.SHARE);
+
+export interface MoreOptionsDropdownProps {
+  children: React.ReactElement;
+  shareUrl?: string;
+  hasMobileUserAgent?: boolean;
+  isEstimated?: boolean;
+}
+
+const dropdownItemStyle = 'flex items-center gap-2 py-2';
+const dropdownContentStyle = 'font-semibold text-xs';
+
+export function MoreOptionsDropdown({
+  children,
+  shareUrl = baseUrl,
+  hasMobileUserAgent = hasMobileUA(),
+  isEstimated = false,
+}: MoreOptionsDropdownProps) {
+  const { t } = useTranslation();
+  const [toastMessage, setToastMessage] = useState('');
+  const { isOpen, onDismiss, onToggleDropdown } = useDropdownCtl();
+  const reference = useToastReference();
+  const { copyToClipboard, share } = useShare();
+
+  const summary = `${t('more-options-dropdown.summary')} ${baseUrl}`;
+
+  const { onShare, copyShareUrl } = useMemo(() => {
```

Comment:
LMK if anyone think this might be a premature optimization 😬

Reply:
I don't think it's premature, if we don't add it from the start we'll probably forget about it 😅